CIPs/cip-granda_mento.md

```diff
@@ -0,0 +1,186 @@
+
+While Granda Mento will help facilitate `CELO <-> stable token` exchanges, this is ultimately meant as a mechanism to help purchasers with fiat (e.g. USD) who are seeking to own Celo's corresponding stable token (e.g. cUSD). Purchasers would come to an agreement with a broker (e.g. the Celo Foundation) who owns an existing amount of CELO. The broker would be the one to make the `CELO -> cUSD` exchange on-chain.
+
+The proposed implementation is being considered in the medium-term for large scale exchanges until Mento is possibly reworked to support more efficient large-scale minting on a sustained basis.
+
+## Specification
+
+At a high level, the design involves:
+1. Anyone can submit an exchange proposal.
+   * The amount of stable tokens being bought/sold in the exchange is required to be within a governable range.
+   * The assets being sold (ie either the stable token or CELO) in the exchange are deposited.
+   * The current oracle price for the exchange is recorded.
+2. The proposed exchange must be approved by a multisig that has previously been authorized by Governance. At this point, the exchange still cannot be executed.
+3. A forced waiting period of X days must elapse before the exchange can be executed. During this time, Governance can choose to veto the exchange, refunding the exchange proposer.
+4. After the waiting period, the exchange is executed according to the oracle price recorded in step 1.
+
+An exchange can have the following states:
+1. Proposed - The exchange has been proposed, but not yet approved by the approver. The proposer may still cancel their proposal and be refunded their deposit.
+2. Approved - The exchange has been proposed and approved by the approver, but not yet executed or vetoed. The proposer may not cancel their proposal.
+3. Executed - The exchange has been proposed, approved by the approver, and executed.
+4. Cancelled - The exchange has been cancelled by the proposer when it was in the Proposed state, or it has been vetoed by Governance when it was in the Approved stage.
+
+### Specific Changes:
+
+#### Modifications of existing contracts:
+
+##### `StableToken.sol`
+
+Currently, `StableToken.sol` only allows its `Exchange.sol` or `Validators.sol` to [mint](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L225) new stable tokens. Similarly, only its `Exchange.sol` can [burn](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L275) stable tokens.
+
+The following modifications will be made:
+1. **Modification to [mint()](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L224)**
+   * Allow `msg.sender` to be the stable token's `GrandaMento` in addition to the existing permitted senders, Exchange and `Validators.sol`.
```

Comment:
interesthing if this should be enabled for al StableTokens or we would like to enable it ad-hoc for cEUR or  other exchanges later on.

Reply:
I feel like StableTokenX should have a governable list of contracts allowed to burn/mint.

Reply:
This line used to make more sense when I had a 1 GrandaMento per StableToken, but I recently moved it to 1 GrandaMento for all StableTokens (fixing now). The current proposal would be to have all StableTokens allow GrandaMento to mint/burn, but the only way for GrandaMento to actually mint/burn is if governance explicitly whitelists a specific StableToken inside GrandaMento

Re: governable list of contracts, I'm a little torn-- I see the value in there being a governable list of contracts, but at the same time I worry adding this feature could up the complexity/surface area of the proposed change. @yorhodes thoughts?

Reply:
From the implementation perspective doesn't seem that complex to make contracts able to allow a particular contract to burn/mint an it can be used in future use cases. I'm not saying we should do it that way, not 100% sold myself, but I think that the decision should be around better UX rather than code simplicity.